source/extensions/filters/network/redis_proxy/command_splitter_impl.cc

```diff
@@ -900,6 +901,293 @@ void MSETRequest::onChildResponse(Common::Redis::RespValuePtr&& value, uint32_t
   }
 }
 
+Common::Redis::Client::PoolRequest* 
+ makeScanRequest(const RouteSharedPtr& route, int32_t shard_index, 
+                Common::Redis::RespValue& incoming_request, 
+                ConnPool::PoolCallbacks& callbacks,
+                Common::Redis::Client::Transaction& transaction) {
+  std::string key = std::string();
+  Extensions::NetworkFilters::RedisProxy::ConnPool::InstanceImpl* req_instance =
+        dynamic_cast<Extensions::NetworkFilters::RedisProxy::ConnPool::InstanceImpl*>(
+            route->upstream(key).get());
+  auto handler = req_instance->makeRequestNoKey(shard_index, ConnPool::RespVariant(incoming_request),
+                                                       callbacks, transaction);
+  return handler;
+}
+
+bool requiresValue(const std::string& arg) {
+    return (arg == "count" || arg == "match" || arg == "type");
+}
+
+// Generic function to build an array with bulkstring
+void addBulkString(Common::Redis::RespValue& requestArray, const std::string& value) {
+    Common::Redis::RespValue element;
+    element.type(Common::Redis::RespType::BulkString);
+    element.asString() = value;
+    requestArray.asArray().emplace_back(std::move(element));
+}
+
+SplitRequestPtr ScanRequest::create(Router& router, Common::Redis::RespValuePtr&& incoming_request,
+                                    SplitCallbacks& callbacks, CommandStats& command_stats,
+                                    TimeSource& time_source, bool delay_command_latency,
+                                    const StreamInfo::StreamInfo& stream_info) {
+
+  // SCAN looks like: SCAN cursor [MATCH pattern] [COUNT count] [TYPE type]
+  // Ensure there are at least two args to the command or it cannot be scanned.
+  // Also the number of arguments should be in even number, otherwise command is invalid
+  if (incoming_request->asArray().size() < 2 || incoming_request->asArray().size() % 2 != 0) {
+    onWrongNumberOfArguments(callbacks, *incoming_request);
+    command_stats.error_.inc();
+    return nullptr;
+  }
+
+  std::string key = std::string();
+  int32_t shard_idx = 0; 
+  int32_t numofRequests= 3;
+  std::string count = "2";
```

Comment:
Increase the count value, later this needs to be a configurable option from protobuf

Reply:
Updated to 1000, fixed for count < 1000 and cached the count value